File: src/test/java/no/nav/fo/veilarbregistrering/registrering/bruker/RegistreringTilstandTestdataBuilder.java
package no.nav.fo.veilarbregistrering.registrering.bruker;

import java.time.LocalDateTime;

public class RegistreringTilstandTestdataBuilder {

    public static Builder registreringTilstand() {
        return new Builder();
    }

    public static class Builder {

        private long brukerRegistreringId = 112233L;
        private LocalDateTime opprettet = LocalDateTime.now();
        private Status status = Status.MOTTATT;
        private LocalDateTime sistEndret = null;

        public RegistreringTilstand build() {
            return RegistreringTilstand.of(
                    -1L,
                    brukerRegistreringId,
                    opprettet,
                    sistEndret,
                    status);
        }

        public Builder brukerRegistreringId(long brukerRegistreringId) {
            this.brukerRegistreringId = brukerRegistreringId;
            return this;
        }

        public Builder opprettet(LocalDateTime opprettet) {
            this.opprettet = opprettet;
            return this;
        }

        public Builder status(Status status) {
            this.status = status;
            return this;
        }

        public Builder sistEndret(LocalDateTime sistEndret) {
            this.sistEndret = sistEndret;
            return this;
        }
    }
}

File: src/test/kotlin/no/nav/fo/veilarbregistrering/registrering/formidling/RegistreringTilstandTestdataBuilder.kt
package no.nav.fo.veilarbregistrering.registrering.formidling

import java.time.LocalDateTime

object RegistreringTilstandTestdataBuilder {

    @JvmStatic
    fun registreringTilstand() = Builder()

    class Builder {
        private var brukerRegistreringId = 112233L
        private var opprettet = LocalDateTime.now()
        private var status = Status.MOTTATT
        private var sistEndret: LocalDateTime? = null
        fun build(): RegistreringTilstand {
            return RegistreringTilstand.of(
                -1L,
                brukerRegistreringId,
                opprettet,
                sistEndret,
                status
            )
        }

        fun brukerRegistreringId(brukerRegistreringId: Long): Builder {
            this.brukerRegistreringId = brukerRegistreringId
            return this
        }

        fun opprettet(opprettet: LocalDateTime): Builder {
            this.opprettet = opprettet
            return this
        }

        fun status(status: Status): Builder {
            this.status = status
            return this
        }

        fun sistEndret(sistEndret: LocalDateTime?): Builder {
            this.sistEndret = sistEndret
            return this
        }
    }
}
